### PR TITLE
Add switch_browser toMicrosoftStsAuthorizationRequest, Fixes AB#3079799

### PR DESCRIPTION
**Pull request by p3dr0rv**
[AB#3079799](https://identitydivision.visualstudio.com/fac9d424-53d2-45c0-91b5-ef6ba7a6bf26/_workitems/edit/3079799)

MicrosoftStsAuthorizationRequest is the object used to represent the request parameters on an authorization request, here we add the new property to represent the parameter switch_browser that will be sent to STS when DUNA is enabled.
https://github.com/AzureAD/ad-accounts-for-android/pull/3012

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 vNext
 ----------
+- [PATCH] Add switch_browser toMicrosoftStsAuthorizationRequest (#2550)
```
Comment (shahzaibj) [inline changelog.txt]:
```suggestion
- [MINOR] Add switch_browser toMicrosoftStsAuthorizationRequest (#2550)
```

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java [hunk 2]
```diff
@@ -186,6 +194,8 @@ public static class Builder extends MicrosoftAuthorizationRequest.Builder<Micros
         private Map<String, String> mFlightParameters = new HashMap<>();
         private OpenIdProviderConfiguration mOpenIdProviderConfiguration;
 
+        private String mSwitchBrowser;
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java:197]: This is meant to be a query param, right? How/where is this converted to qp?

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java [hunk 3]
```diff
@@ -125,6 +125,13 @@ public class MicrosoftStsAuthorizationRequest extends MicrosoftAuthorizationRequ
     // TODO private transient InstanceDiscoveryMetadata mInstanceDiscoveryMetadata;
     // TODO private boolean mIsExtendedLifetimeEnabled = false;
 
+    // The value 1 indicates that client supports DUNA flow.
+    @Getter
+    @Accessors(prefix = "m")
+    @SerializedName("switch_browser")
+    private final String mSwitchBrowser;
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java:132]: This is meant to be a query parameter, right? How/where is this converted to qp?

Comment (p3dr0rv) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java:132]: yes, this qp is sent in the auth request.
the default value is null meaning that when converted to string will not be added.
Only when duna is enabled we set this to 1
https://github.com/AzureAD/ad-accounts-for-android/blob/4a0b3736b8086b86c4e19a45fb13d55efcbc14e8/broker4j/src/main/com/microsoft/identity/broker4j/broker/prt/AbstractPrtAuthorizationStrategy.java#L100

**Review by shahzaibj — APPROVED** [review]

**Review by rpdome — APPROVED** [review]

**Review by mohitc1 — APPROVED** [review]
:shipit:

**Pull request merged**